src/volatile_ptr/operations.rs

```diff
@@ -82,6 +82,7 @@ where
     /// };
     /// assert_eq!(pointer.read(), 42);
     /// ```
+    #[must_use]
```

Comment:
A user might be calling this function to cause side-effects and not necessarily because they're interested in the value.

Reply:
Ah, that's true.

For those cases, it might still be beneficial to be explicit about wanting side effects by discarding the value:

```rust
let _side_effect = pointer.read();
```

I can remove it, if you want me to, though. I would prefer `#[must_use]` and be explicit about side effects, but I have no strong opinion on this.